### PR TITLE
LPS-166239 Adapt value store and retrieve for MultiselectPicklist Field

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-166239

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContext.java [hunk 1]
```diff
@@ -1068,6 +1069,25 @@ else if (value instanceof ListEntry) {
...
+			for (int i = 0; i < listEntries.size(); i++) {
+				ListEntry listEntry = listEntries.get(i);
+
+				sb.append(listEntry.getKey());
+
+				if (i < (listEntries.size() - 1)) {
+					sb.append(StringPool.COMMA_AND_SPACE);
+				}
+			}
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContext.java:1086]: @p-albuquerque can we simplify to something like:

```
com.liferay.petra.string.StringUtil.merge(
       ListUtil.toList(listEntries, ListEntry::getKey), 
       StringPool.COMMA_AND_SPACE);
```
?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 2]
```diff
@@ -58,7 +58,10 @@ public static long getListTypeDefinitionId(
 
 		if (!StringUtil.equals(
 				objectField.getBusinessTypeAsString(),
-				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST) &&
+			!StringUtil.equals(
+				objectField.getBusinessTypeAsString(),
+				ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:64]: @p-albuquerque could you sort this alphabetically? i.e., `BUSINESS_TYPE_MULTISELECT_PICKLIST` before `BUSINESS_TYPE_PICKLIST`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -2271,9 +2271,23 @@ private void _insertIntoTable(
...
+						objectField.getBusinessType(),
+						ObjectFieldConstants.
+							BUSINESS_TYPE_MULTISELECT_PICKLIST)) {
+
+					String stringValue = String.valueOf(
+						values.get(objectField.getName()));
+
+					_setColumn(
+						preparedStatement, index++, column.getSQLType(),
+						stringValue.substring(1, stringValue.length() - 1));
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2284]: @p-albuquerque can we change to something like:

```
StringUtil.removeChars(stringValue, CharPool.OPEN_BRACKET, CharPool.CLOSE_BRACKET)
```
to make it more intuitive? or the key can have these characters?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -2268,12 +2268,9 @@ private void _insertIntoTable(
...
 				}
 
-				Column<?, ?> column = dynamicObjectDefinitionTable.getColumn(
-					objectField.getDBColumnName());
+				_setColumn(index, objectField, preparedStatement, values);
 
-				_setColumn(
-					preparedStatement, index++, column.getSQLType(),
-					values.get(objectField.getName()));
+				index++;
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2273]: @p-albuquerque can we pass the `index++` directly like we were doing before?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 5]
```diff
@@ -2461,6 +2458,35 @@ private void _reindex(ObjectEntry objectEntry) throws PortalException {
...
+			_setColumn(
+				preparedStatement, index, column.getSQLType(),
+				StringUtil.removeChars(
+					stringValue, CharPool.OPEN_BRACKET,
+					CharPool.CLOSE_BRACKET));
+		}
+		else {
+			_setColumn(
+				preparedStatement, index, column.getSQLType(),
+				values.get(objectField.getName()));
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2486]: @p-albuquerque can we move the `values.get(objectField.getName())` out as follows:

```
Column<?, ?> column = dynamicObjectDefinitionTable.getColumn(
    objectField.getDBColumnName());

Object value = values.get(objectField.getName());

if (StringUtil.equals(
        objectField.getBusinessType(),
        ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {

    _setColumn(
        preparedStatement, index, column.getSQLType(),
        StringUtil.removeChars(
            String.valueOf(value), CharPool.OPEN_BRACKET,
            CharPool.CLOSE_BRACKET));
}
else {
    _setColumn(
        preparedStatement, index, column.getSQLType(), value);
}
```

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java [hunk 6]
```diff
@@ -137,6 +138,29 @@ private DTOConverterContext _getDTOConverterContext(
 			dtoConverterContext.getUser());
 	}
 
+	private ListEntry _getListEntry(
+		long listTypeDefinitionId, DTOConverterContext dtoConverterContext,
+		String key) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java:143]: @p-albuquerque can we sort this alphabetically?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Review by danvb112 — APPROVED** [review]
Frontend side LGTM!

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:reevaluate:1277401_5593

**Comment by carolmariaabb** [comment]
@p-albuquerque The failures seem unrelated to your PR. 

Feel free to send this PR directly to Brian.

**Comment by p-albuquerque** [comment]
Resent in: https://github.com/brianchandotcom/liferay-portal/pull/125365

**Pull request closed without merging**